core/drivers/firewall/stm32_etzpc.c

```diff
@@ -372,6 +409,211 @@ stm32_etzpc_acquire_memory_access(struct firewall_query *firewall,
 	return TEE_ERROR_ACCESS_DENIED;
 }
 
+#ifdef CFG_STM32MP15
+static bool pager_permits_decprot_config(uint32_t decprot_id,
+					 enum etzpc_decprot_attributes attr)
+{
+	paddr_t ram_base = 0;
+	size_t ram_size = 0;
+
+	if (!IS_ENABLED(CFG_WITH_PAGER))
+		return true;
+
+	switch (decprot_id) {
+	case ETZPC_TZMA1_ID:
+		ram_base = SYSRAM_BASE;
+		ram_size = SYSRAM_SEC_SIZE;
+		break;
+	case STM32MP1_ETZPC_SRAM1_ID:
+		ram_base = SRAM1_BASE;
+		ram_size = SRAM1_SIZE;
+		break;
+	case STM32MP1_ETZPC_SRAM2_ID:
+		ram_base = SRAM2_BASE;
+		ram_size = SRAM2_SIZE;
+		break;
+	case STM32MP1_ETZPC_SRAM3_ID:
+		ram_base = SRAM3_BASE;
+		ram_size = SRAM3_SIZE;
+		break;
+	case STM32MP1_ETZPC_SRAM4_ID:
+		ram_base = SRAM4_BASE;
+		ram_size = SRAM4_SIZE;
+		break;
+	default:
+		return true;
+	}
+
+	if (stm32mp1_ram_intersect_pager_ram(ram_base, ram_size) &&
+	    attr != ETZPC_DECPROT_S_RW) {
+		EMSG("Internal RAM %#"PRIxPA"..%#"PRIxPA" is used by pager, shall be secure",
+		     ram_base, ram_base + ram_size);
+		return false;
+	}
+
+	return true;
+}
+#endif /* CFG_STM32MP15 */
+
+static bool decprot_id_is_sram(uint32_t id)
```

Comment:
Name doesn't really match as there is retram for mp15.
I'd switch to `decprot_id_is_tzma_ram()` and change accordingly to recognize the 2 TZMA IDs

Reply:
ok, i'll rename to `decprot_id_is_internal_ram()`